go-controller/pkg/ovn/egressfirewall_test.go

```diff
@@ -692,9 +692,266 @@ var _ = ginkgo.Describe("OVN EgressFirewall Operations for local gateway mode",
+				}
+				nodeSwitch2 := &nbdb.LogicalSwitch{
+					UUID: node2Name + "-UUID",
+					Name: node2Name,
+				}
+
+				dbSetup := libovsdbtest.TestSetup{
+					NBData: []libovsdbtest.TestData{
+						nodeSwitch1,
+						nodeSwitch2,
+						clusterRouter,
+					},
+				}
 
+				namespace1 := *newNamespace("namespace1")
+				egressFirewall := newEgressFirewallObject("default", namespace1.Name, []egressfirewallapi.EgressFirewallRule{
+					{
+						Type: "Allow",
+						Ports: []egressfirewallapi.EgressFirewallPort{
+							{
+								Protocol: "TCP",
+								Port:     100,
+							},
+						},
+						To: egressfirewallapi.EgressFirewallDestination{
+							CIDRSelector: "1.2.3.5/23",
+						},
+					},
+				})
+
+				fakeOVN.startWithDBSetup(dbSetup,
+					&egressfirewallapi.EgressFirewallList{
+						Items: []egressfirewallapi.EgressFirewall{
+							*egressFirewall,
+						},
+					},
+					&v1.NodeList{
+						Items: []v1.Node{
+							{
+								Status: v1.NodeStatus{
+									Phase: v1.NodeRunning,
+								},
+								ObjectMeta: newObjectMeta(node1Name, ""),
+							},
+							{
+								Status: v1.NodeStatus{
+									Phase: v1.NodeRunning,
+								},
+								ObjectMeta: newObjectMeta(node2Name, ""),
+							},
+						},
+					})
+
+				fakeOVN.controller.WatchEgressFirewall()
+
+				ipv4ACL := libovsdbops.BuildACL(
+					"",
+					t.DirectionToLPort,
+					t.EgressFirewallStartPriority,
+					"(ip4.dst == 1.2.3.5/23) && ip4.src == $a10481622940199974102 && ((tcp && ( tcp.dst == 100 ))) && ip4.dst != 10.128.0.0/14",
+					nbdb.ACLActionAllow,
+					"",
+					"",
+					false,
+					map[string]string{"egressFirewall": "namespace1"},
+					nil,
+				)
+				ipv4ACL.UUID = "ipv4ACL-UUID"
+
+				// new ACL will be added to the switches
+				nodeSwitch1.ACLs = []string{ipv4ACL.UUID}
+				nodeSwitch2.ACLs = []string{ipv4ACL.UUID}
+
+				expectedDatabaseState := []libovsdb.TestData{
+					ipv4ACL,
+					nodeSwitch1,
+					nodeSwitch2,
+					clusterRouter,
+				}
+
+				gomega.Expect(fakeOVN.nbClient).To(libovsdbtest.HaveData(expectedDatabaseState))
+
+				ginkgo.By("Bringing down NBDB")
+				// inject transient problem, nbdb is down
+				fakeOVN.controller.nbClient.Close()
+				gomega.Eventually(func() bool {
+					return fakeOVN.controller.nbClient.Connected()
+				}).Should(gomega.BeFalse())
+
+				err := fakeOVN.fakeClient.EgressFirewallClient.K8sV1().EgressFirewalls(egressFirewall.Namespace).Delete(context.TODO(), egressFirewall.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				// sleep long enough for TransactWithRetry to fail, causing egress firewall Add to fail
+				time.Sleep(t.OVSDBTimeout + time.Second)
+				// check to see if the retry cache has an entry for this egress firewall
+				key := getEgressFirewallNamespacedName(egressFirewall)
+				gomega.Eventually(func() *retryObjEntry {
+					return fakeOVN.controller.retryEgressFirewalls.getObjRetryEntry(key)
+				}).ShouldNot(gomega.BeNil())
```

Comment:
Here a delete operation failed, so we should check that:
- there is a retry entry (as you did)
- `entry.oldObj != nil`, which means that we will try deleting this egress firewall again